Units/parser-javascript.r/1795612.js.d/expected.tags

```diff
@@ -1,4 +1,4 @@
-RPC	input.js	/^Test.RPC =$/;"	c	class:Test
-asyncMethod	input.js	/^	asyncMethod: function($/;"	m	class:Test.RPC
-asyncRequest	input.js	/^	asyncRequest: function($/;"	m	class:Test.RPC
-request_id	input.js	/^	request_id: 0,$/;"	p	class:Test.RPC
+Test.RPC	input.js	/^Test.RPC =$/;"	p
```

Comment:
Is this change expected?  Admittedly `Test` is not actually declared in this file so creating a tag for it is a bit artificial, but this change looks a bit unexpected (it's at least not obvious to me which change did this, so I can't tell if it was intentional).

I don't really mind either way as there's no definite answer (it's not even proper JS given `Test` is not even imported or anything), but I'm wondering if this was on purpose or an accident.

Reply:
This looks like a bug about the granularity of tagging.

Looking at this again, a tag (`Test.RPC`) with more than one token (`Test` and `RPC`) looks strange. (as far as comparing with 
Units/parser-javascript.r/js-class-related-unterminated.d/expected.tags)

My expectation (--extras=+{reference} --fields=+{roles}{scope}):

```
RPC	input.js	/^Test.RPC =$/;"	p	scope:variable:Test
Test	input.js	/^Test.RPC =$/;"	v	roles:chainElt
```

I introduced the `chainElt` roles for avoiding more-than-one-token tag entries.


 
There is room for discussion about the kinds (p for `RPC` and v for `Test`), but there will be no perfect answer. It is up to the parser maintainer.

Reply:
I'll convert this to an issue.

Reply:
```diff
diff --git a/parsers/jscript.c b/parsers/jscript.c
index 3788cc090..f54c432da 100644
--- a/parsers/jscript.c
+++ b/parsers/jscript.c
@@ -485,7 +485,7 @@ static int makeJsTagCommon (const tokenInfo *const token, const jsKind kind,
 
 	const char *p;
 	char *name_chain = NULL;
-	if (!token->dynamicProp && kind != JSTAG_PROPERTY &&  (p = strrchr (name, '.')) != NULL )
+	if (!token->dynamicProp && (p = strrchr (name, '.')) != NULL )
 	{
 		if ((p - name) != 0)
 			name_chain = eStrndup (name, (size_t) (p - name));
```

With this, we can avoid the two-tokens tags.
It is easy to fix the parser as we want it, but it is not easy to decide how the parser should be.

Reply:
We should open two issues. I will open issues for them.